conclave-common/src/main/kotlin/com/r3/conclave/common/internal/attestation/AttestationJsonTypes.kt

```diff
@@ -423,10 +511,84 @@ enum class EnclaveTcbStatus {
 /**
  * @property isvsvn SGX Enclave’s ISV SVN.
  */
-data class EnclaveTcb @JsonCreator constructor(
-    @JsonProperty("isvsvn")
-    val isvsvn: Int
-)
+data class EnclaveTcb constructor(val isvsvn: Int) {
+    companion object {
+        fun fromJson(json: JsonNode): EnclaveTcb {
+            return EnclaveTcb(json.getInt("isvsvn"))
+        }
+    }
+}
+
+/**
+ * Utility functions for manual parsing of json primitives and producing sensible error messages.
+ */
+private fun JsonNode.checkFieldExists(fieldName: String) {
+    check(this.has(fieldName)) { "No such field $fieldName." }
+}
+
+private fun JsonNode.getInt(fieldName: String): Int {
+    checkFieldExists(fieldName)
+    val field = this.get(fieldName)
+    check(field.isInt) { "Expected $fieldName to be an integer, got ${field.nodeType}." }
+    return field.asInt()
+}
+
+private fun JsonNode.getString(fieldName: String): String {
+    checkFieldExists(fieldName)
+    val field = this.get(fieldName)
+    check(field.isTextual) { "Expected $fieldName to be a string, got ${field.nodeType}." }
+    return field.asText()
+}
+
+private fun JsonNode.getHexEncodedBytes(fieldName: String): OpaqueBytes {
+    val stringField = getString(fieldName)
+    val errorMessage by lazy { "$fieldName is not a valid hex string, " }
+    check(stringField.length % 2 == 0) { "$errorMessage, number of characters must be even." }
+    return try {
+        val bytes = stringField.chunked(2)
+                .map { it.toInt(16).toByte() }
+                .toByteArray()
+        OpaqueBytes(bytes)
+    } catch (e: NumberFormatException) {
+        throw IllegalArgumentException("$errorMessage, invalid characters.")
+    }
+}
+
+private fun JsonNode.getInstant(fieldName: String, format: String, timezone: String): Instant {
+    val dateString = getString(fieldName)
+    return try {
+        val dateFormat = SimpleDateFormat(format).apply { timeZone = TimeZone.getTimeZone(timezone) }
```

Comment:
Use `DateTimeFormatter`, which is the newer API

Reply:
Wasn't able to get DateTimeFormatter to parse the format correctly. Switched to Instant.parse() instead.